src/jit/codegenarmarch.cpp

```diff
@@ -1359,6 +1359,11 @@ void CodeGen::genCodeForIndir(GenTreeIndir* tree)
     genConsumeAddress(tree->Addr());
     emit->emitInsLoadStoreOp(ins_Load(targetType), emitTypeSize(tree), targetReg, tree);
     genProduceReg(tree);
+
+    if (tree->gtFlags & GTF_IND_VOLATILE)
+    {
+        instGen_MemoryBarrier();
+    }
```

Comment:
> Is there an improved variant that could be used on ARM64 that a full memory barrier?

I started with a port of the arm32 legacy JIT to arm64 RyuJIT because I wanted to keep the review simple.

Based on your comment that these barriers are not needed on x64/x86....  I think there are some things we can do
+ Use `dmb st` instead of `dmb sy`
+ Only insert memory barrier before a store 

I think that would make functionally equivalent to x64.


+ The ld acquire and st release forms may be useful, but that will take a bit more time for me to 
understand the IL semantics with respect to the Barrier semantics.